docs/hooks.md

```diff
@@ -0,0 +1,291 @@
+# Hooks
+
+In this example, we'll build a simplistic React app showing a list of colors to see how one could integrate **Baobab** with React by using hooks.
+
+### Summary
+
+* [Hooks](#hooks)
+* [Summary](#summary)
+* [Creating the app's state](#creating-the-apps-state)
+* [Rooting our top-level component](#rooting-our-top-level-component)
+* [Branching our list](#branching-our-list)
+* [Actions](#actions)
+* [Dynamically set the list's path using props](#dynamically-set-the-lists-path-using-props)
+* [Clever vs. dumb components](#clever-vs-dumb-components)
+
+### Creating the app's state
+
+Let's create a **Baobab** tree to store our colors' list:
+
+*state.js*
+
+```js
+import Baobab from 'baobab';
+
+const tree = new Baobab({
+  colors: ['yellow', 'blue', 'orange']
+});
+
+export default tree;
+```
+
+### Rooting our top-level component
+
+Now that the tree is created, we should bind our React app to it by "rooting" our top-level component.
+
+Under the hood, this component will simply propagate the tree to its descendants using React's [Context](https://reactjs.org/docs/context.html) so that "branched" component may subscribe to updates of parts of the tree afterwards.
+
+*main.jsx*
+
+```jsx
+import React, {Component} from 'react';
+import {render} from 'react-dom';
+import {useRoot} from 'baobab-react/hooks';
+import tree from './state';
+
+// We will write this component later
+import List from './list.jsx';
+
+// Creating our top-level component
+const App = function({tree}) {
```

Comment:
Could we avoid variable scope shadowing here, related to the `tree` name?

Reply:
We could pass it down as `store`?